test/code/plugins/lib_patch_management_test.rb

```diff
@@ -0,0 +1,288 @@
+    @inventoryPath = File.join(File.dirname(__FILE__), 'InventoryWithUpdates.xml')
+    LinuxUpdates.prev_hash = nil
+    LinuxUpdates.log = OMS::MockLog.new
+  end
+
+  def teardown
+
+  end
+
+  def test_strToXML
+    xml = LinuxUpdates.strToXML(@installed_packages_xml_str)
+    assert(xml.is_a?(REXML::Document), "Expected return type is REXML::Document")
+  end
+
+  def test_strToXML_fail
+    assert_raise REXML::ParseException do
+      LinuxUpdates.strToXML("<<<<")
+    end
+  end
+  
+  def test_getAvailableUpdatesInstancesXML
+    xml = LinuxUpdates.strToXML(@available_updates_xml_str)
+    assert(xml.root != nil, 'Failed find the root of the xml document')
+    assert_equal("INSTANCE", xml.root.name)
+    instances = LinuxUpdates.getInstancesXML(xml)
+    
+    assert_equal(1, instances.size)
+    assert_equal("INSTANCE", instances[0].name)
+    assert_equal("MSFT_nxAvailableUpdatesResource", instances[0].attributes['CLASSNAME'])
+  end
+
+  def test_availableUpdatesXMLtoHash
+    instanceXMLstr = '{
+      <INSTANCE CLASSNAME="MSFT_nxAvailableUpdatesResource">
+        <PROPERTY NAME="Name" TYPE="string">
+            <VALUE>dpkg</VALUE>
+        </PROPERTY>
+        <PROPERTY NAME="Version" TYPE="string">
+            <VALUE>1.18.4ubuntu1.1</VALUE>
+        </PROPERTY>
+        <PROPERTY NAME="Architecture" TYPE="string">
+            <VALUE>amd64</VALUE>
+        </PROPERTY>
+        <PROPERTY NAME="Repository" TYPE="string">
+            <VALUE>Ubuntu:16.04/xenial-updates</VALUE>
+        </PROPERTY>
+        <PROPERTY NAME="BuildDate" TYPE="string">
+            <VALUE>BuildDate</VALUE>
+        </PROPERTY>
+      </INSTANCE>'
+    
+    expectedHash = {
+      "CollectionName"=> "dpkg" + @@delimiter + "1.18.4ubuntu1.1" + @@delimiter + 
+                                @@hostOSDetailsMap["OSFullName"],
+      "PackageName"=>"dpkg",
+      "PackageVersion"=>"1.18.4ubuntu1.1",
+      "Timestamp"=> "1970-01-01T00:00:00.000Z",
+      "Repository"=>"Ubuntu:16.04/xenial-updates",
+      "Installed"=>false,
+    }
+    
+    instanceXML = LinuxUpdates::strToXML(instanceXMLstr).root
+    assert_equal("INSTANCE", instanceXML.name)
```

Comment:
Sure. I'll change the asserts to more elaborate messages. In this case, the instanceXML.name is literally searching for the work "INSTANCE".